### ISSUE TITLE
Decimal 18 token

### ISSUE THREAD

**Issue by ChihChengLiang** [thehubbleproject/hubble-contracts#475]
### What's wrong

The floating-point compression we use now has 4 bits for the exponent (and 12 bits for the mantissa), which supports up to a  10^15 value.

So if we have a new token with decimals 18 registered, which is the most common number for decimal, we'll have some problem.
Take [UNI](https://etherscan.io/tx/0xf82d8e769b67dda9c3992bca9fafb61a8e9f5c296672869c75cb917feca2aadc) for example, 
- To send 1 UNI, we use mantissa 1000 and all the exponent bits.
- The maximum UNI we can send in one transfer is 4.095 UNI
- But we are allowed to send 10^-18 UNI, which is not a common use case.

### How can we fix it

In the old discussion(#208), we fix the Ether case by denoting Ether in Gwei. We can do the same trick for the ERC20 tokens. But the Ether case means

- We register Ether as Gwei in tokenRegistry.
- We interpret the state balance as Gwei. If someone deposits 1 ETH, then the deposit manager creates a leaf of balance 1*10^9, instead of 1 * 10^18.
- Some conversion in the withdraw is required too. If someone withdraws 1 Gwei (integer 1) from the state leaf, the withdraw manager should send 1*10^9 wei to them. 

It would be more clear if we change the above example in WETH token. What we need to do is 

- The tokenRegistry registers a `new_decimal` 9. We also define a `diff_decimal` = `original_decimal` - `new_decimal` = 9
- 1 WETH in the state balance is 1* 10^9, which is 1* 10^(`new_decimal`)
- If someone sends 1 WETH (1*10^18) to the deposit manager, it divides the original number with the 10^(`diff_decimal`) to get the correct state balance to create.
- To withdraw the 1 WETH in that state leaf, the withdraw manager multiplies the state leaf value with 10^(`diff_decimal`) and sends  1*10^18 to the user.

The question is what's the policy we should define the new_decimal for all ERC20 tokens? 
- We can define a rule that applies to all decimal cases. But I haven't figured it out yet.
- We register token manually and define new_decimal by case. 
  - [Almost all token](https://wispy-bird-88a7.uniswap.workers.dev/?url=http://tokens.1inch.eth.link) is decimal 18 with the most significant exception of USDT series having decimal 6. Some other minor tokens have decimal less than 18.
  - We currently don't have a clear idea who can/should introduce a new token in the token registry. If we don't like to introduce complicated governance, we just assume the tokens are predefined when the system is deployed.

**Comment by ChihChengLiang** [comment]
"We can define a rule that applies to all decimal cases. "
```
require(erc20.decimal <= 18, "Sorry you are strange don't serve")
if erc20.decimal >= 9:
    new_decimal = erc20.decimal - 9
else:
    new_decimal = erc20.decimal
```

**Comment by ChihChengLiang** [comment]
Fixed in #478
